src/throttle/amoc_throttle_config.erl

```diff
@@ -0,0 +1,188 @@
+    PoolConfig#{1 := First#{status => active}};
+pool_config(Rate, Interval) ->
+    NoOfProcesses = no_of_processes(),
+    RatePerMinutePerProcess = (60000 * Rate div Interval) div NoOfProcesses,
+    DelayPerProcess = (NoOfProcesses * Interval) div Rate,
+    Rem = ((60000 * Rate div Interval) rem NoOfProcesses)
+            + ((NoOfProcesses * Interval) rem Rate),
+    calculate_availability(RatePerMinutePerProcess, DelayPerProcess, NoOfProcesses, Rem).
+
+-spec process_pool_config(pid(), pool_config()) -> pool_config().
+process_pool_config(PoolSup, PoolConfig) ->
+    Processes = supervisor:which_children(PoolSup),
+    Workers = [ {N, Pid} || {{amoc_throttle_process, N}, Pid, _, _} <- Processes, is_pid(Pid) ],
+    Fun1 = fun(N, Config) ->
+                   {_, Pid} = lists:keyfind(N, 1, Workers),
+                   Config#{pid => Pid}
+           end,
+    maps:map(Fun1, PoolConfig).
+
+-spec calculate_availability(integer(), integer(), pos_integer(), integer()) -> pool_config().
+calculate_availability(RatePerMinutePerProcess, DelayPerProcess, NoOfProcesses, Rem) ->
+    Fun = fun(N, {Acc, R}) ->
+                  case {RatePerMinutePerProcess < NoOfProcesses, R} of
+                      {true, 0} ->
+                          Config = #{max_n => RatePerMinutePerProcess,
+                                     delay => DelayPerProcess + 1,
+                                     status => inactive, pid => undefined},
+                          {Acc#{N => Config}, R};
+                      {true, R} ->
+                          Config = #{max_n => RatePerMinutePerProcess,
+                                     delay => DelayPerProcess,
+                                     status => active, pid => undefined},
+                          {Acc#{N => Config}, R - 1};
+                      {false, 0} ->
+                          Config = #{max_n => RatePerMinutePerProcess,
+                                     delay => DelayPerProcess,
+                                     status => active, pid => undefined},
+                          {Acc#{N => Config}, R};
+                      {false, R} ->
+                          Config = #{max_n => RatePerMinutePerProcess,
+                                     delay => DelayPerProcess + 1,
+                                     status => active, pid => undefined},
+                          {Acc#{N => Config}, R - 1}
+                  end
+          end,
+    {#{} = PoolConfig, _} = lists:foldl(Fun, {#{}, Rem}, lists:seq(1, NoOfProcesses)),
+    PoolConfig.
+
+-spec no_of_processes() -> non_neg_integer().
+no_of_processes() ->
+    3 * erlang:system_info(schedulers_online).
+
+-spec do_verify_gradual_config(amoc_throttle:gradual_rate_config()) -> gradual().
+do_verify_gradual_config(
+  #{from_interarrival := FromInterarrival, to_interarrival := ToInterarrival} = Config0)
+  when ?POS_INT(FromInterarrival), ?POS_INT(ToInterarrival) ->
```

Comment:
when ?POS_INT(FromInterarrival), ?POS_INT(ToInterarrival), not is_map_key(interval, Config0) ->